packages/core-data/src/resolvers.js

```diff
@@ -293,6 +324,18 @@ export const getEntityRecords =
 	};
 
 getEntityRecords.shouldInvalidate = ( action, kind, name ) => {
+	// Invalidate cache when a new revision is created.
```

Comment:
I don't understand the change here, why the revisions are invalidated using a different action than the records. For instance why is this not using `RECEIVE_ITEM_REVISIONS`

Reply:
I might not be understanding the invalidation flow, but I thought it was logical that we'd want to invalidate the revisions cache when a new post has been saved, not after having received revisions items, by when it's too late.

I also did it this way because the `shouldInvalidate` method wasn't receiving args that matched the action. 

 I think it's because when the `RECEIVE_ITEM_REVISIONS` action is fired `getEntityRecords` resolver is still resolving for some reason and doesn't  [satisfy the middleware check](https://github.com/WordPress/gutenberg/blob/cebb4ada8e6a1cc86b154f0810e8e15779aaa7d0/packages/data/src/resolvers-cache-middleware.js#L32-L32).

Having said that, I can retest all this after adding dedicated resolvers as per your [comment above](https://github.com/WordPress/gutenberg/pull/54046/files#r1372792888). It's likely I missed something.

Reply:
I think you might be right here, there might be more things that should invalidate the revisions and since saving a post create revisions, it's a good one to add.